### ISSUE TITLE
Subtyping checker does not support `@PolyAll`

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#2081]
First, consider [section 24.2.3 of the manual ("The @PolyAll qualifier applies to every type system")](https://checkerframework.org/manual/#qualifier-polymorphism). The title of the subsection is pretty clear that `@PolyAll` should apply to _all_ type systems. The text in the section doesn't clarify the issue further - it just gives the justification for `@PolyAll`.

The Subtyping Checker is one of the simplest checkers supported by the framework, and a common entry point for new users wishing to build their own checkers. However, it does not support `@PolyAll`, even if the user writes a polymorphic qualifier - `@PolyAll` qualifiers are silently ignored. 

To reproduce this, run the subtyping checker using the Tainting Checker's hierarchy, like this:
`javac -processor org.checkerframework.common.subtyping.SubtypingChecker -Aquals=org.checkerframework.checker.tainting.qual.Tainted,org.checkerframework.checker.tainting.qual.Untainted,org.checkerframework.checker.tainting.qual.PolyTainted PolyAllFail.java`.

(Any simple hierarchy with a polymorphic type will do, but tainting was convenient.)

Here is a simple demonstration of the failure (`PolyAllFail.java`):
```java
import org.checkerframework.checker.tainting.qual.*;
import org.checkerframework.framework.qual.PolyAll;

public class PolyAllFail {
  @PolyTainted Object id_taint(@PolyTainted Object obj) { return obj; }

  @PolyAll Object id_all(@PolyAll Object obj) { return obj; }

  void demo(@Untainted Object obj) {
     @Untainted Object objT = id_taint(obj);
     @Untainted Object objA = id_all(obj);
  }
}
```

Expected: no errors
Actual:
```
PolyAllFail.java:11: error: [assignment.type.incompatible] incompatible types in assignment.
	@Untainted Object objA = id_all(obj);
	                               ^
  found   : @Tainted Object
  required: @Untainted Object
1 error
```

I believe the problem is that `SubtypingAnnotatedTypeFactory#createSupportedTypeQualifiers` doesn't name `@PolyAll` explicitly.
